### PR TITLE
#30 [5.x branch] Add a possibility to define the search query when adding visualizations

### PR DESCRIPTION
**Pull request by Boris-Toninski-OrangeDot**
Added only to the 5.x branch.

### CODE REVIEW

**Comment by ytzlax** [comment]
Thank you for help me improve this plugin, I will look on this PR in the next days

File: public/test/kibana-api.spec.js [hunk 1]
```diff
@@ -184,6 +184,16 @@ describe('KibanaApiService', () => {
             res = KibanaApiService["getKibanaDocumentStructure"]("title", {a: "abc"}, "myIndex");
             expect(res.visState).equal(JSON.stringify({a: "abc"}))
         });
+        it('Should return object with query equal to * by default', () => {
+            res = KibanaApiService["getKibanaDocumentStructure"]("title", {}, "myIndex");
+            var searchSourceJSON = JSON.parse(res.kibanaSavedObjectMeta.searchSourceJSON);
+            expect(searchSourceJSON.query.query_string.query).equal("*");
+        });
```
Comment (ytzlax) [inline public/test/kibana-api.spec.js]:
```
  it('Should return object with query equal to {match_all: {}} by default', () => {
            res = KibanaApiService["getKibanaDocumentStructure"]("title", {}, "myIndex");
            var searchSourceJSON = JSON.parse(res.kibanaSavedObjectMeta.searchSourceJSON);
            expect(searchSourceJSON.query).to.deep.equal({match_all: {}});
        });
```

File: public/test/kibana-api.spec.js [hunk 2]
```diff
@@ -184,6 +184,16 @@ describe('KibanaApiService', () => {
...
+        it('Should return object with query equal to * by default', () => {
+            res = KibanaApiService["getKibanaDocumentStructure"]("title", {}, "myIndex");
+            var searchSourceJSON = JSON.parse(res.kibanaSavedObjectMeta.searchSourceJSON);
+            expect(searchSourceJSON.query.query_string.query).equal("*");
+        });
+        it('Should return object with query equal to "_type: a AND type: b"', () => {
+            res = KibanaApiService["getKibanaDocumentStructure"]("title", {}, "myIndex", "_type: a AND type: b");
+            var searchSourceJSON = JSON.parse(res.kibanaSavedObjectMeta.searchSourceJSON);
+            expect(searchSourceJSON.query.query_string.query).equal("_type: a AND type: b");
+        });
```
Comment (ytzlax) [inline public/test/kibana-api.spec.js]:
```
  it('Should return object with query_string equal to "_type: a AND type: b"', () => {
            let query = {
                query_string: {
                    analyze_wildcard: true,
                    query: "_type: a AND type: b"
                }
            }
            res = KibanaApiService["getKibanaDocumentStructure"]("title", {}, "myIndex", query);
            var searchSourceJSON = JSON.parse(res.kibanaSavedObjectMeta.searchSourceJSON);
            expect(searchSourceJSON.query).to.deep.equal(query);
        });
```

File: public/kibana-api-service.js [hunk 3]
```diff
@@ -130,13 +130,15 @@ export class KibanaApiService {
...
      * @returns {{title: *, visState, uiStateJSON: string, description: string, kibanaSavedObjectMeta: {searchSourceJSON}}}
      */
-    static getKibanaDocumentStructure(iTitle, iVisState, iIndex) {
+    static getKibanaDocumentStructure(iTitle, iVisState, iIndex, query) {
+
         let kibanaSavedObjectMeta = {
             searchSourceJSON: {
                 index: iIndex,
-                query: {query_string: {query: "*", analyze_wildcard: true}},
+                query: {query_string: {query: query || "*", analyze_wildcard: true}},
```
[the remark this answers is not in the document]
Comment (Boris-Toninski-OrangeDot) [reply public/kibana-api-service.js]: Sorry, for not answering till now, I was really busy.

I agree, with the change, but have a question.

Initially the default value of the query was "{query_string: {query: "*", analyze_wildcard: true}}", but now you suggests to change it to "{match_all: {}}". Are the above two equivalent?

[the remark this answers is not in the document]
Comment (ytzlax) [reply public/kibana-api-service.js]: Hi,
You are right, previously it was  "{query_string: {query: "*", analyze_wildcard: true}}", but its not true, its better to be {match_all: {}}

[the remark this answers is not in the document]
Comment (Boris-Toninski-OrangeDot) [reply public/kibana-api-service.js]: Hi, 
I just pushed the change you suggested.

**Comment by ytzlax** [comment]
I just merge the changes.
If you like this repository, please give a star
thanks

**Comment by ytzlax** [comment]
I will update the 5.6.0 version release and above in the next days, I will let you know

**Pull request merged**